File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int n = numbers.size();
        int i = 0;
        int j = n-1;
// Two ppointer Algorithm
        while(i<j){
            if(numbers[i]+numbers[j] > target){
                j--;
            }
            else if(numbers[i]+numbers[j] < target){
                i++;
            }
            else{
                return{i+1,j+1};           // 1 based indexing ke liye i+1; j+1
            }
        }
        return{};
    }

};

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        n = len(numbers)
        i = 0
        j = n-1
        while i < j :
            if numbers[i]+numbers[j] > target:
                j-=1
            elif numbers[i]+numbers[j] < target:
                i+=1    
            else:
                return[i+1, j+1]  

        return []        